tests/Support/SupportCollectionTest.php

```diff
@@ -865,6 +865,21 @@ public function testTransform()
         $this->assertEquals(['first' => 'first-rolyat', 'last' => 'last-llewto'], $data->all());
     }
 
+    public function testTranspose()
+    {
+        $data = new Collection([
+            'name' => ['Tyrion Lannister', 'Daenerys Targaryen', 'Meera Reed'],
+            'house' => ['Lannister', 'Targaryen', 'Stark'],
+            'actor' => ['Peter Dinklage', 'Emilia Clarke', 'Ellie Kendrick'],
+        ]);
+        $expected = [
+            ['Tyrion Lannister', 'Lannister', 'Peter Dinklage'],
+            ['Daenerys Targaryen', 'Targaryen', 'Emilia Clarke'],
+            ['Meera Reed', 'Stark', 'Ellie Kendrick'],
+        ];
+        $this->assertEquals($expected, $data->transpose()->toArray());
```

Comment:
Trying to think of the best way to document that the inner arrays will actually be collections (since we're gonna make that switch).

`toArray` is sort of a big hammer in the sense that it tries to convert everything to an array recursively, including anything that implements `Arrayable` like an Eloquent model.

Since `assertEquals` does a loose comparison, this actually works and I think might be the best way to explicitly define the expected output:

```php
        $expected = new Collection([
            new Collection(['Tyrion Lannister', 'Lannister', 'Peter Dinklage']),
            new Collection(['Daenerys Targaryen', 'Targaryen', 'Emilia Clarke']),
            new Collection(['Meera Reed', 'Stark', 'Ellie Kendrick']),
        ]);
```